### ISSUE TITLE
Mouse offset from actual window (ChromeOS)

### ISSUE THREAD

**Issue by lheinold** [alacritty/alacritty#8351]
### System

OS: ChromeOS (Version 130.0.6723.126)
Version: alacritty 0.15.0-dev (8731d157)
Linux/BSD: Wayland

### Logs

Font/Terminal size: `alacritty -vv`
```
Created log file at "/tmp/Alacritty-13901.log"
[0.000005755s] [INFO ] [alacritty] Welcome to Alacritty
[0.000281768s] [INFO ] [alacritty] Version 0.15.0-dev (8731d157)
[0.000316076s] [INFO ] [alacritty] Running on Wayland
[0.000802558s] [INFO ] [alacritty] Configuration files loaded from:
                                     "/home/heinold/.config/alacritty/alacritty.toml"
[0.033557407s] [INFO ] [alacritty] Using EGL 1.4
[0.033638690s] [DEBUG] [alacritty] Picked GL Config:
                                     buffer_type: Some(Rgb { r_size: 8, g_size: 8, b_size: 8 })
                                     alpha_size: 8
                                     num_samples: 0
                                     hardware_accelerated: true
                                     supports_transparency: Some(true)
                                     config_api: Api(OPENGL | GLES1 | GLES2 | GLES3)
                                     srgb_capable: true
[0.048033751s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Bold, load_flags: LoadFlag(TARGET_LIGHT), render_mode: "Normal", lcd_filter: 1 }
[0.051163193s] [INFO ] [alacritty] Window scale factor: 1.6
[0.051844609s] [DEBUG] [alacritty] Loading "MesloLGS Nerd Font Mono" font
[0.052704155s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Regular, load_flags: LoadFlag(TARGET_LIGHT), render_mode: "Normal", lcd_filter: 1 }
[0.054050582s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Bold, load_flags: LoadFlag(TARGET_LIGHT), render_mode: "Normal", lcd_filter: 1 }
[0.055371332s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Italic, load_flags: LoadFlag(TARGET_LIGHT), render_mode: "Normal", lcd_filter: 1 }
[0.056760353s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Bold Italic, load_flags: LoadFlag(TARGET_LIGHT), render_mode: "Normal", lcd_filter: 1 }
[0.059797222s] [INFO ] [alacritty] Running on virgl (Mesa Intel(R) Graphics (RPL-U))
[0.059835859s] [INFO ] [alacritty] OpenGL version 4.3 (Core Profile) Mesa 22.3.6, shader_version 4.30
[0.059850876s] [INFO ] [alacritty] Using OpenGL 3.3 renderer
[0.069658931s] [DEBUG] [alacritty] Enabled debug logging for OpenGL
[0.069731320s] [DEBUG] [alacritty] Filling glyph cache with common glyphs
[0.072377949s] [INFO ] [alacritty] Cell size: 14 x 31
[0.072410316s] [INFO ] [alacritty] Padding: 0 x 0
[0.072416640s] [INFO ] [alacritty] Width: 1280, Height: 960
[0.072467988s] [INFO ] [alacritty] PTY dimensions: 30 x 91
[0.074434488s] [INFO ] [alacritty] Initialisation complete
[0.081176338s] [INFO ] [alacritty] Font size changed to 24.0 px
[0.081229051s] [INFO ] [alacritty] Cell size: 14 x 31
[0.083270060s] [INFO ] [alacritty] Padding: 0 x 0
[0.083290032s] [INFO ] [alacritty] Width: 1280, Height: 960
```
Keyboard and bindings: `alacritty --print-events`
```
[0.000005410s] [INFO ] [alacritty] Welcome to Alacritty
[0.000156204s] [INFO ] [alacritty] Version 0.15.0-dev (8731d157)
[0.000165663s] [INFO ] [alacritty] Running on Wayland
[0.000332441s] [INFO ] [alacritty] Configuration files loaded from:
                                     "/home/heinold/.config/alacritty/alacritty.toml"
[0.031172649s] [INFO ] [alacritty] Using EGL 1.4
[0.046873523s] [INFO ] [alacritty] Window scale factor: 1.6
[0.055053948s] [INFO ] [alacritty] Running on virgl (Mesa Intel(R) Graphics (RPL-U))
[0.055113230s] [INFO ] [alacritty] OpenGL version 4.3 (Core Profile) Mesa 22.3.6, shader_version 4.30
[0.055125995s] [INFO ] [alacritty] Using OpenGL 3.3 renderer
[0.068788959s] [INFO ] [alacritty] Cell size: 14 x 31
[0.068868022s] [INFO ] [alacritty] Padding: 0 x 0
[0.068873046s] [INFO ] [alacritty] Width: 1280, Height: 960
[0.068922665s] [INFO ] [alacritty] PTY dimensions: 30 x 91
[0.070996726s] [INFO ] [alacritty] Initialisation complete
[0.071029413s] [INFO ] [alacritty_winit_event] ScaleFactorChanged { scale_factor: 1.6, inner_size_writer: InnerSizeWriter { new_inner_size: (Weak) } }
[0.071039697s] [INFO ] [alacritty_winit_event] Resized(PhysicalSize { width: 1280, height: 960 })
[0.075935268s] [INFO ] [alacritty_winit_event] RedrawRequested
[0.078002958s] [INFO ] [alacritty] Font size changed to 24.0 px
[0.078039265s] [INFO ] [alacritty] Cell size: 14 x 31
[0.080092256s] [INFO ] [alacritty] Padding: 0 x 0
[0.080104599s] [INFO ] [alacritty] Width: 1280, Height: 960
[0.080401118s] [INFO ] [alacritty_winit_event] About to wait
[0.080453830s] [INFO ] [alacritty_winit_event] About to wait
[0.082694227s] [INFO ] [alacritty_winit_event] Event { window_id: Some(WindowId(94713223610448)), payload: Terminal(Wakeup) }
[0.082759935s] [INFO ] [alacritty_winit_event] About to wait
[0.082778242s] [INFO ] [alacritty_winit_event] About to wait
[0.109772001s] [INFO ] [alacritty_winit_event] Focused(true)
[0.109866589s] [INFO ] [alacritty_winit_event] ModifiersChanged(Modifiers { state: ModifiersState(0x0), pressed_mods: ModifiersKeys(0x0) })
[0.109886128s] [INFO ] [alacritty_winit_event] About to wait
[0.110588455s] [INFO ] [alacritty_winit_event] About to wait
[0.113773607s] [INFO ] [alacritty_winit_event] RedrawRequested
[0.115091846s] [INFO ] [alacritty_winit_event] About to wait
[0.876694407s] [INFO ] [alacritty_winit_event] About to wait
[0.887165310s] [INFO ] [alacritty_winit_event] CursorEntered { device_id: DeviceId(Wayland(DeviceId)) }
[0.887240208s] [INFO ] [alacritty_winit_event] CursorMoved { device_id: DeviceId(Wayland(DeviceId)), position: PhysicalPosition { x: 1279.3500000000001, y: 298.09375 } }
[0.887272855s] [INFO ] [alacritty_winit_event] About to wait
```

On ChromeOS, the mouse position seems to be offset from the actual window. For example, instead of resizing by clicking near the window border, you have to click way below the border in order to resize. You also have to click below the title bar buttons to minimize/maximize/close, or to move the window.

Videos: 
[Screen recording 2024-12-10 4.21.48 PM.webm](https://github.com/user-attachments/assets/3adc4286-74e9-42d7-9a27-aea35b102da5)
[Screen recording 2024-12-10 4.23.10 PM.webm](https://github.com/user-attachments/assets/b6b33cc8-6da1-43ca-ad0e-d85b1cc4de38)

This seems similar to #3191 except I'm on the latest version.

EDIT:
Did a little bit more digging and it looks like this is linked to the screen magnification setting. The default on my chromebook is 1600x1000 (which the settings say is 100% magnification). When I change it to the smallest possible magnification (native resolution) it is 2560x1600 (63% magnification in the settings). At this resolution alacritty reports the window scale factor to be 1.0, and the offset problem goes away. The more I increase the magnification, the bigger the offset becomes. Unfortunately, I have bad eyesight and the smallest magnification is pretty much unreadable to me, so keeping it there isn't really an option.

**Comment by chrisduerr** [comment]
Non-1 scales definitely work with Alacritty, including fractional scaling. If @kchibisov thinks there might be a Wayland issue here, please feel free to reopen, but considering that this only seems to be an issue on ChromeOS I'm not interested to put any resources towards this.

**Comment by kchibisov** [comment]
That really sounds like ChromeOs issue. Also, is the offset present for the actual terminal itself, like when you select text?

**Comment by lheinold** [comment]
It is, unfortunately - it's actually how I first noticed it, because when you try to select text it's so offset it selects the wrong line. Super annoying.

Interestingly, Terminator seems to work fine - all Linux apps on ChromeOS use Wayland so I started comparing the Wayland debug logs next to see if anything sticks out. Only thing I've noticed so far is that Terminator doesn't use `wp_fractional_scale_manager_v1`, which Alacritty/winit does. My current theory is that the scale factor returned by ChromeOS isn't what winit/Wayland is expecting which is causing issues? I tested fractional scaling on Wayland + Ubuntu and (as expected) it works fine.

Unfortunately ChromeOS is super limited on customization options so I don't see a way to turn fractional scaling off. ChromeOS is far from my first choice for development but this is for work so I'm trying to find ways to fix the (numerous) issues I'm running into.
